src/linalg/nonbacktracking.jl

```diff
@@ -26,17 +26,22 @@ function non_backtracking_matrix(g::AbstractGraph)
         end
     end
 
-    B = zeros(Float64, m, m)
-
+    nz = if is_directed(g)
+        sum(indegree(g) .* outdegree(g))
```

Comment:
this allocates needlessly

Reply:
Do you mean that it is better not to assign the value to `nz`?

Reply:
I mean that `indegree(g)` and `outdegree(g)` are vectors, so you allocate two vectors plus a third one for their elementwise product, just to compute the sum

Reply:
you can compute it in a for loop with zero allocation:
```julia
nz = sum(indegree(g, v) * outdegree(g, v) for v in vertices(g))
```

Reply:
I appreciate your suggestion. Thank you very much!